modules/compute/function-app/main.bicep

```diff
@@ -0,0 +1,314 @@
+
+@description('Optional. The kind of application that this component refers to, used to customize UI.')
+param appInsightsKind string = 'azfunc'
+
+@description('Optional. Enabled or Disable Insights for Azure functions')
+param enableInsights bool = false
+
+@description('Optional. Resource ID of the log analytics workspace which the data will be ingested to, if enableaInsights is false.')
+param workspaceResourceId string = ''
+
+@description('Optional. List of Azure function (Actual object where our code resides).')
+param functions array = []
+
+@description('Optional. Enable Vnet Integration or not.')
+param enableVnetIntegration bool = false
+
+@description('Optional. The subnet that will be integrated to enable vnet Integration.')
+param subnetId string = ''
+
+@description('Optional. Enable Source control for the function.')
+param enableSourceControl bool = false
+
+@description('Optional. Repository or source control URL.')
+param repoUrl string = ''
+
+@description('Optional. Name of branch to use for deployment.')
+param branch string = ''
+
+@description('Optional. to limit to manual integration; to enable continuous integration (which configures webhooks into online repos like GitHub).')
+param isManualIntegration bool = true
+
+@description('Optional. true for a Mercurial repository; false for a Git repository.')
+param isMercurial bool = false
+
+@description('Required. Name of the storage account used by function app.')
+param storgeAccountName string
+
+@description('Required. Resource Group of storage account used by function app.')
+param storgeAccountResourceGroup string
+
+@description('Optional. True to deploy functions from zip package. "functionPackageUri" must be specified if enabled. The package option and sourcecontrol option should not be enabled at the same time.')
+param enablePackageDeploy bool = false
+
+@description('Optional. URI to the function source code zip package, must be accessible by the deployer. E.g. A zip file on Azure storage in the same resource group.')
+param functionPackageUri string = ''
+
+@description('''Optional. Extra app settings that should be provisioned while creating the function app. Note! Settings below should not be included unless absolutely necessary, because settings in this param will override the ones added by the module:
+AzureWebJobsStorage
+AzureWebJobsDashboard
+WEBSITE_CONTENTSHARE
+WEBSITE_CONTENTAZUREFILECONNECTIONSTRING
+FUNCTIONS_EXTENSION_VERSION
+FUNCTIONS_WORKER_RUNTIME
+WEBSITE_NODE_DEFAULT_VERSION
+APPINSIGHTS_INSTRUMENTATIONKEY
+APPLICATIONINSIGHTS_CONNECTION_STRING''')
+param extraAppSettings object = {}
+
+
+
+
+@description('Defines storageAccounts for Azure Function App.')
+resource storageAccount 'Microsoft.Storage/storageAccounts@2021-06-01' existing = {
+  name: storgeAccountName
+  scope: resourceGroup(storgeAccountResourceGroup)
+}
+
+@description('Defines Application service plan.')
+resource serverfarms 'Microsoft.Web/serverfarms@2021-02-01' = {
```

Comment:
Is there a reason to be using an API versions throughout?

Reply:
Hosting plan is an optional requirement, but it's required if you run your function on a premium plan , it was added as dependency